src/index.native.scroll.js

```diff
@@ -103,7 +103,9 @@ class SwipeableViews extends React.Component {
       index,
     } = nextProps;
 
-    if (typeof index === 'number' && index !== this.props.index) {
+    // Sometimes the index can be the same, but the slide
+    // may have changed, so just always set it
+    if (typeof index === 'number') {
```

Comment:
@oliviertassinari That's what I'd have thought, but for some reason it doesn't get updated unless the index is directly set. Happy to dig around a bit more and see if there's a better way, unless you have any thoughts?